openapi-spec/swagger.yaml

```diff
@@ -532,11 +516,254 @@ paths:
+          content:
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/BaseResponse'
+      requestBody:
+        content:
+          application/xml:
+            schema:
+              $ref: '#/components/schemas/CreateBucketConfiguration'
+    delete:
+      tags:
+        - s3
+      summary: delete a bucket.
+      operationId: bucketdelete
+      responses:
+        '200':
+          description: successful delete bucket
+          content:
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/BaseResponse'
+  v1/s3:
+    parameters:
+      - name: bucketName
+        in: path
+        description: the name of bucket
+        required: true
+        schema:
+          type: string
+    get:
+      tags:
+        - s3
+      summary: list buckets.
+      operationId: listBuckets
+      responses:
+        '200':
+          description: successful delete bucket
+          content:
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/ListAllMyBucketsResult'
+  'v1/s3/{bucketName}/{object}':
+    parameters:
+      - name: bucketName
+        in: path
+        description: the name of bucket
+        required: true
+        schema:
+          type: string
+      - name: object
+        in: path
+        description: the name of object
+        required: true
+        schema:
+          type: string
+      - name: Range
+        in: path
+        description: >-
+          the Header parameter Range value like "bytes:1-10",used for multipart
+          download
+        required: false
+        schema:
+          type: string
+    put:
+      tags:
+        - s3
+      summary: upload a object.
+      operationId: objectPut
+      responses:
+        '200':
+          description: successful upload a object
+          content:
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/BaseResponse'
+    get:
+      tags:
+        - s3
+      summary: download object
+      operationId: objectGet
+    delete:
+      tags:
+        - s3
+      summary: delete a object.
+      operationId: objectdelete
+      responses:
+        '200':
+          description: successful delete object
+          content:
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/BaseResponse'
+  'v1/s3/{bucketName}/':
+    parameters:
+      - name: bucketName
+        in: path
+        description: the name of bucket
+        required: true
+        schema:
+          type: string
+    get:
+      tags:
+        - s3
+      summary: list objects
+      operationId: listObjects
+      responses:
+        '200':
+          description: successful list objects
+          content:
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/ListObjectResponse'
+  'v1/s3/{bucketName}/{object}?uploads':
```

Comment:
If I get it right, this should be changed into `v1/s3/{bucketName}/{object}/uploads`.